Quran/QuranViewController.swift

```diff
@@ -193,6 +204,45 @@ class QuranViewController: UIViewController, AudioBannerViewPresenterDelegate {
         setBarsHidden(navigationController?.navigationBarHidden == false)
     }
 
+    func onViewLongTapped(sender: UILongPressGestureRecognizer) {
+        guard sender.state == UIGestureRecognizerState.Ended else { return }
```

Comment:
1. This code assumes that the image is full screen which is not the case. Actually, the only class responsible for laying out the image is `QuranPageCollectionViewCell` and accordingly these calculations should be there and using `mainImageView.bounds.size` and `mainImageView.image.size` with no hard coded values.

2. Loading an image is async. Sometimes the screen becomes blank and we shouldn't handle long press there. So, I think the gesture recognizer needs to move as well to `QuranPageCollectionViewCell` with a delegate handling `gestureRecognizerShouldBegin` with `mainImageView.image.image != nil`.